integration/integration_test.go

```diff
@@ -5457,3 +5459,61 @@ func TestTraitsPropagation(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "hello leaf", strings.TrimSpace(outputLeaf))
 }
+
+// testSessionStreaming tests streaming events from session recordings.
+func testSessionStreaming(t *testing.T, suite *integrationTestSuite) {
+	ctx := context.Background()
+	sessionID := session.ID(uuid.New())
+	teleport := suite.newTeleport(t, nil, true)
+	defer teleport.StopAll()
+
+	api := teleport.GetSiteAPI(Site)
+	auditStream, err := api.CreateAuditStream(ctx, sessionID)
+	require.Nil(t, err)
+
+	for i := 0; i < 1000; i++ {
+		err = auditStream.EmitAuditEvent(ctx, &apievents.SessionPrint{
+			Metadata: apievents.Metadata{
+				Index: int64(i),
+				Type:  events.SessionPrintEvent,
+				Time:  time.Now(),
+			},
+			ChunkIndex: int64(i),
+		})
+
+		require.Nil(t, err)
+	}
+
+	err = auditStream.Complete(ctx)
+	require.Nil(t, err)
+	start := time.Now()
+
+	for {
+		sessionPlayback, e := api.StreamSessionEvents(ctx, sessionID, 0)
+		startEvent := <-sessionPlayback
+		_, ok := startEvent.(*apievents.SessionStart)
+		require.True(t, ok)
+
+		for i := 0; i < 1000; i++ {
+			select {
+			case event := <-sessionPlayback:
+				printEvent, ok := event.(*apievents.SessionPrint)
+				require.True(t, ok)
+				require.Equal(t, i, printEvent.ChunkIndex)
+			case <-ctx.Done():
+				require.Nil(t, ctx.Err())
+			case err := <-e:
+				if time.Since(start) < time.Minute*5 {
+					time.Sleep(time.Second * 5)
+					continue
+				}
```

Comment:
I don't understand, are you expecting to get more events after an error? Wouldn't the `continue` here mess up your loop counter if you're still expecting 1000 events?

Reply:
Yeah I need to add a loop label so it continues the outer loop. This is needed to retry because uploading a stream with complete isn’t synchronous. Same goes for below comment. Forgot to push that commit

Reply:
Reworked the test. Apologies